types/braintree/index.d.ts

```diff
@@ -0,0 +1,1749 @@
+// Type definitions for braintree 2.16
+// Project: https://github.com/braintree/braintree_node
+// Definitions by: Sam Rubin <https://github.com/smrubin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+export = braintree;
+export as namespace braintree;
+
+declare namespace braintree {
+    /**
+     * Braintree Config and Client
+     */
+
+    export enum Environment {
+        Production,
+        Sandbox
+    }
+
+    export interface Config {
+        environment: Environment;
+        merchantId: string;
+        publicKey: string;
+        privateKey: string;
+    }
+
+    export class BraintreeGateway {
+        constructor(config: Config);
+        addOn: AddOnGateway;
+        address: AddressGateway;
+        clientToken: ClientTokenGateway;
+        creditCard: CreditCardGateway;
+        creditCardVerification: CreditCardVerificationGateway;
+        customer: CustomerGateway;
+        discount: DiscountGateway;
+        dispute: DisputeGateway;
+        merchantAccountGateway: MerchantAccountGateway;
+        paymentMethod: PaymentMethodGateway;
+        paymentMethodNonce: PaymentMethodNonceGateway;
+        plan: PlanGateway;
+        settlementBatchSummary: SettlementBatchSummaryGateway;
+        subscription: SubscriptionGateway;
+        testing: TestingGateway;
+        transaction: TransactionGateway;
+        transactionLineItem: TransactionLineItemGateway;
+    }
+
+    type Entity = 'address' | 'creditCard' | 'customer' | 'dispute' | 'merchantAccount' | 'paymentMethod' | 'settlementBatchSummary' | 'subscription' | 'transaction';
+
+    type ValidatedResponse<T> = {
```

Comment:
Still a little weird to me that technically this means that `'address': CreditCard` could be valid then. Personal preference, have the Entity mapping.

Reply:
Agreed, implemented your suggestion!